node/libs/roles/src/validator/tests.rs

```diff
@@ -254,12 +261,13 @@ fn test_prepare_qc() {
                 &setup1.genesis,
             );
         }
-        if n >= setup1.genesis.validators.threshold() {
+        let expected_weight = n as u64 * setup1.genesis.validators.total_weight() / 6;
+        if expected_weight >= setup1.genesis.validators.threshold() {
             qc.verify(&setup1.genesis).unwrap();
         } else {
             assert_matches!(
                 qc.verify(&setup1.genesis),
-                Err(Error::NotEnoughSigners { .. })
```

Comment:
nit: IMO NotEnoughSigners is more comprehensive name than WeightNotReached. All that changes is the definition of "enough".